rfcs/text/0030-linux-event-model.md

```diff
@@ -0,0 +1,1299 @@
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+This RFC aims to introduce a significant number of additions to ECS (mostly the 'Process' fieldset) in an effort to provide a rich enough linux event model to drive Session view in Kibana, as well as provide wider context to EQL and other rule engines. Care has been taken to ensure this RFC doesn't break any existing uses of the process field set, and should be fully backwards compatible with existing endpoint agents.
+
+<!--
+Stage 1: If the changes include field additions or modifications, please create a folder titled as the RFC number under rfcs/text/. This will be where proposed schema changes as standalone YAML files or extended example mappings and larger source documents will go as the RFC is iterated upon.
+-->
+
+<!--
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+## Fields
+
+<!--
+Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+<!--
+Stage 2: Add or update all remaining field definitions. The list should now be exhaustive. The goal here is to validate the technical details of all remaining fields and to provide a basis for releasing these field definitions as beta in the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+## Usage
+
+<!--
+Stage 1: Describe at a high-level how these field changes will be used in practice. Real world examples are encouraged. The goal here is to understand how people would leverage these fields to gain insights or solve problems. ~1-3 paragraphs.
+-->
+
+1. Session View
+
+The primary use case for this data will be to drive the "Session view". A time ordered rendering of a linux session process tree. The addition of nested process ancestors entry_leader, session_leader, and group_leader will allow Session View to load only the events for a particular session.
+
+KQL examples:
+process.entry_leader.entity_id: &lt;entity_id of entry session leader>
+
+or
+
+process.session_leader.entity_id: &lt;entity_id of session leader>
+
+group_leader process info as well as file descriptors (process.fds) will allow the Session View to properly represent pipes and redirects in a familiar (shell) way.
```

Comment:
@threat-punter @joe-desimone and jake (I don't know his github).

I've heard interest in the past for typing together processes across pipes for EQL rules. With this RFC's spec finally make that a reality?

Reply:
By capturing 'pipes' in our array of fds, we can use the inodes for the pipes to accurately represent the order in which commands were piped together. This info works for session view, but I'm not familiar enough with how EQL works to comment on whether it satisfies requirements there.

Reply:
@ferullo yes - the group_leader is part of this, followed by looking at at stdin/fd0 and stdout/fd1 to see if they have the same pipe ino.   Then we can re-create this:
```
cat foo | grep bar | sort 
```

we capture which side of the pipe is the write side vs read side to help with this (e.g. cat's fd1 would have write side of pipe with same ino as grep's stdin/fd0, which would be the read side).